### PR TITLE
Added generic exception mapper

### LINKED ISSUE #586: DataManagementApi: create centralized exception handler
Controller classes should not return individual HTTP Status Codes themselves, thus they must not return `Response` objects. Instead they should return the DTOs directly or throw exceptions.

All exceptions are caught in one place and mapped to HTTP status codes. This can be achieved by implementing the `ExceptionMapper` interface (check [this example](https://mincong.io/2018/12/03/exception-handling-in-jax-rs/)). 

The following mapping should be implemented: 
- 400: `IllegalArgumentException`, should be thrown whenever query params etc. are missing or invalid. May also include `NullpointerException`.
- 401: `AuthorizationService` (Issue #583) throws `AuthorizationFailedException`, no authentication header present
- 403: `AuthorizationService` throws `NotAuthorizedException`, supplied auth header does not have the proper permissions
- 404: `ObjectNotFoundException`
- 409: `ObjectExistsException`, usually when creating an object whos ID already exists
- 423: `ObjectNotModifiableException`, `IllegalStateException`, thrown when e.g. an Asset cannot be deleted anymore, because it's already used in a contract.
- 501: `UnsupportedOperationException`
- 503: any other unhandled exception

### PR DESCRIPTION
**Pull request by paullatzelsperger**
closes #586

### CODE REVIEW

File: extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/model/Transformer.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition.model;
```
Comment (ndr-brt) [inline extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/model/Transformer.java]: Probably not part of this PR

File: extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/EdcApiExtension.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.exception;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+
+//marker class
+public abstract class EdcApiExtension extends EdcException {
```
Comment (ndr-brt) [inline extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/EdcApiExtension.java:20]: Probably a typo, should be called `EdcApiException`

File: extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/mappers/EdcApiExceptionMapper.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.eclipse.dataspaceconnector.api.exception.AuthorizationFailedException;
+import org.eclipse.dataspaceconnector.api.exception.NotAuthorizedException;
+import org.eclipse.dataspaceconnector.api.exception.ObjectExistsException;
+import org.eclipse.dataspaceconnector.api.exception.ObjectNotFoundException;
+import org.eclipse.dataspaceconnector.api.exception.ObjectNotModifiableException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Provider
```
Comment (ndr-brt) [inline extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/mappers/EdcApiExceptionMapper.java:29]: Would be better to register it to the `resourceConfig`explicitly?
It will make this process more readable (those kind of annotation hide details and that's not a great thing)

File: extensions/api/api-core/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (ndr-brt) [inline extensions/api/api-core/build.gradle.kts:1]: This module should be declared as dependency of `control-api` and `observability-api` modules

File: extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/mappers/EdcApiExceptionMapper.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+public class EdcApiExceptionMapper implements ExceptionMapper<Throwable> {
+    private final Map<Class<? extends Throwable>, Integer> exceptionMap;
+
+    public EdcApiExceptionMapper() {
+        exceptionMap = new HashMap<>(Map.of(
```
Comment (ndr-brt) [inline extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/mappers/EdcApiExceptionMapper.java:34]: the `HashMap` wrapping is not needed, `Map.of` already did the job.

File: extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/model/Transformer.java (hunk 1 above)
Comment (paullatzelsperger) [reply extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/model/Transformer.java]: already deleted. was a copy-paste error.

File: extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/EdcApiExtension.java (hunk 2 above)
Comment (paullatzelsperger) [reply extensions/api/api-core/src/main/java/org/eclipse/dataspaceconnector/api/exception/EdcApiExtension.java:20]: of course :D

File: extensions/api/api-core/build.gradle.kts (hunk 4 above)
Comment (paullatzelsperger) [reply extensions/api/api-core/build.gradle.kts:1]: The upcoming feature of configurable jetty contexts (#601) will allow that different APIs are exposed under different ports/context paths, so the `ExceptionMapper` will have to be registered explicitly in every one of these. Declaring the dependency onto `api-core` will only register it for the default context.

Slightly unrelated: `control-api` and `datamanagement-api` (#542) have a considerable overlap, so I think it makes sense to collate them into one API and expose them on the same context.


Also, for the `observability-api` I think this mapper really doesn't provide much value.

**Review by ndr-brt — APPROVED** [review]
